experimental/FTheoryTools/src/AbstractFTheoryModels/methods.jl

```diff
@@ -624,6 +630,39 @@ function set_zero_section(m::AbstractFTheoryModel, desired_value::Vector{String}
   set_attribute!(m, :zero_section => [f(eval_poly(l, R)) for l in desired_value])
 end
 
+function set_gauge_algebra(m::AbstractFTheoryModel, algebras::Vector{String})
+gauge_algebras = [];
+C = algebraic_closure(QQ);
+for g in algebras
+    if g == "0"
+      continue
+    end
+    if g == "u(1)"
+      algebra = lie_algebra(C,1,[C(1im)*identity_matrix(C,1)],["i"]);
+      push!(gauge_algebras, algebra);
+    elseif g[1:2] == "su"
+      algebra = special_linear_lie_algebra(C,eval(Meta.parse(g[4:end-1])));
```

Comment:
```suggestion
      algebra = special_linear_lie_algebra(C,parse(Int, g[4:end-1]));
```
If you just expect integers there and not mathematical expressions, this is wayyy safer as you don't allow arbitrary code execution.

Reply:
This will sometimes contain mathematical expressions as well

Reply:
Along the lines of "2*5+1"

Reply:
To elaborate more:

The `FTheoryTools` database contains "instructions" for infinite family of geometries. One such family has "instructions" that allow to create one geometry for each non-negative integer k. The k-th geometry in this family comes associated with the Lie algebra `su(2k+1)`.

The above line should 1) replace the character/string `k` by a user given integer value (say 5), 2) evaluate the resulting mathematical expression (2*5+1) to an integer (11) and 3) pass that integer to the Lie algebra constructor.

Suggestions for this @lgoettgens ?

Reply:
This is now of the form parse(Int... as the eval(Meta.parse()) is now done elsewhere